### PR TITLE
build(deps): bump helm/kind-action from 1.8.0 to 1.10.0

### PR DESCRIPTION
**Pull request by ChristianGeie**
update pre-build node image checksums

### CODE REVIEW

**Comment by jekkel** [comment]
Be sure to [update the readme](https://github.com/kiwigrid/k8s-sidecar/blob/master/README.md?plain=1#L36), too.

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -43,25 +43,27 @@ jobs:
...
-            digest: sha256:6e2d8b28a5b601defe327b98bd1c2d1930b49e5d8c512e1895099e4504007adb
+            digest: sha256:5d548739ddef37b9318c70cb977f57bf3e5015e4552be4e27e57280a8cbb8e4f
           - maj_min: v1.27
-            digest: sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72
+            digest: sha256:681253009e68069b8e01aad36a1e0fa8cf18bb0ab3e5c4069b2e65cafdd70843
           - maj_min: v1.28
-            digest: sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31            
+            digest: sha256:9bc6c451a289cf96ad0bbaf33d416901de6fd632415b076ab05f5fa7e4f65c58            
+          - maj_min: v1.29
+            digest: sha256:51a1434a5397193442f0be2a297b488b6c919ce8a3931be0ce822606ea5ca245
```
Comment (jekkel) [inline .github/workflows/build_and_test.yaml:58]: I think you used the digests of [kind v0.22.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.22.0) where our [kind action (v1.10.0)](https://github.com/helm/kind-action/releases/tag/v1.10.0) only uses [kind v0.21.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.21.0) which lists different digests.

Comment (ChristianGeie) [reply .github/workflows/build_and_test.yaml:58]: thx for pointing me out. i overlooked the dependenciy between [kind action (v1.10.0)](https://github.com/helm/kind-action/releases/tag/v1.10.0) and [kind v0.21.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.21.0). I'll adjust that.

**Comment by jekkel** [comment]
Last but not least, once merged, we should add v1.28 and v1.29 tests as required checks in the repo config.

**Review by jekkel — APPROVED** [review]

**Comment by ChristianGeie** [comment]
makes sense, i will add a rule for required tests for the `master` branch

**Review by jekkel — APPROVED** [review]

**Comment by ChristianGeie** [comment]
@jekkel fyi: I added the tests for k8s v1.28 and v1.29 to the `master` branch protection rules.

**Pull request merged**